components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -691,6 +676,16 @@ private void callDefaultProvisioningHandler(String username, AuthenticationConte
         localClaimValues.put(FrameworkConstants.ASSOCIATED_ID,
                 stepConfig.getAuthenticatedUser().getAuthenticatedSubjectIdentifier());
         localClaimValues.put(FrameworkConstants.IDP_ID, stepConfig.getAuthenticatedIdP());
+
+        /*
+        If TOTP is enabled for federated users, the initial federated user login will be identified with the following
+        check and will set the secret key claim for the federated user who is going to be provisioned.
+         */
+        if (context.getProperty(FrameworkConstants.SECRET_KEY_CLAIM_URL) != null) {
```

Comment:
can't we put from authenticator level?

Reply:
In framework, claims are resolved for JIT provisioned user through claim mappings and if there is no any claim mappings only it will get the attributes from the authenticated user.